annoying but real failure:

```
======================================================================
FAIL: Suite should include remote-request if searching is configured
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_suite_remote_request.py", line 56, in test_remote_request
    self.assertXmlPartialEqual(self.get_xml('remote_request'), suite, "./remote-request[1]")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 26, in assertXmlPartialEqual
    self.assertXmlEqual(expected, actual, normalize=False)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -10,10 +10,10 @@
         </text>
       </display>
     </command>
+    <instance id="groups" src="jr://fixture/user-groups"/>
     <instance id="locations" src="jr://fixture/commtrack:locations"/>
     <instance id="casedb" src="jr://instance/casedb"/>
     <instance id="commcaresession" src="jr://instance/session"/>
-    <instance id="groups" src="jr://fixture/user-groups"/>
     <session>
       <query storage-instance="results" template="case" url="https://www.example.com/a/test_domain/phone/search/">
         <data key="case_type" ref="'case'"/>
----------------------------------------------------------------------
```

--- a/corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js
+++ b/corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js
@@ -219,12 +219,26 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
             var self = this;
             self.name = ko.observable(name);
             self.label = ko.observable(label);
+
+            self.name.subscribe(function () {
+                saveButton.fire('change');
+            });
+            self.label.subscribe(function () {
+                saveButton.fire('change');
+            });
         };
 
         var DefaultProperty = function (property, defaultValue) {
             var self = this;
             self.property = ko.observable(property);
             self.defaultValue = ko.observable(defaultValue);
+
+            self.property.subscribe(function () {
+                saveButton.fire('change');
+            });
+            self.defaultValue.subscribe(function () {
+                saveButton.fire('change');
+            });
         };
 
         self.relevant = ko.observable();
@@ -245,9 +259,6 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
         } else {
             self.searchProperties.push(new SearchProperty('', ''));
         }
-        self.searchProperties.subscribe(function () {
-            saveButton.fire('change');
-        });
 
         self.addProperty = function () {
             self.searchProperties.push(new SearchProperty('', ''));
@@ -281,9 +292,6 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
         } else {
             self.defaultProperties.push(new DefaultProperty('', ''));
         }
-        self.defaultProperties.subscribe(function () {
-            saveButton.fire('change');
-        });
         self.addDefaultProperty = function () {
             self.defaultProperties.push(new DefaultProperty('',''));
         };
@@ -323,10 +331,17 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
                 default_properties: self._getDefaultProperties(),
             };
         };
+
         self.includeClosed.subscribe(function () {
             saveButton.fire('change');
         });
         self.default_relevant.subscribe(function () {
+            saveButton.fire('change');
+        });
+        self.searchProperties.subscribe(function () {
+            saveButton.fire('change');
+        });
+        self.defaultProperties.subscribe(function () {
             saveButton.fire('change');
         });
     };

--- a/corehq/apps/app_manager/suite_xml/post_process/instances.py
+++ b/corehq/apps/app_manager/suite_xml/post_process/instances.py
@@ -57,11 +57,19 @@ class EntryInstances(PostProcessor):
                     for datum in frame.datums:
                         xpaths.add(datum.value)
         xpaths.discard(None)
-        instances, unknown_instance_ids = EntryInstances.get_required_instances(xpaths)
-
-        instances, unknown_instance_ids = self._add_custom_referenced_instances(instances, unknown_instance_ids)
+        instances, unknown_instance_ids = EntryInstances.get_all_instances(self.app.domain, xpaths)
 
         entry.require_instances(instances=instances, instance_ids=unknown_instance_ids)
+
+    @staticmethod
+    def get_all_instances(domain, xpaths):
+        instances, unknown_instance_ids = EntryInstances.get_required_instances(xpaths)
+        instances, unknown_instance_ids = EntryInstances._add_custom_referenced_instances(
+            domain,
+            instances,
+            unknown_instance_ids
+        )
+        return instances, unknown_instance_ids
 
     @staticmethod
     def get_required_instances(xpaths):
@@ -88,21 +96,22 @@ class EntryInstances(PostProcessor):
                     unknown_instance_ids.add(instance_name)
         return instances, unknown_instance_ids
 
-    def _add_custom_referenced_instances(self, instances, unknown_instance_ids):
+    @staticmethod
+    def _add_custom_referenced_instances(domain, instances, unknown_instance_ids):
         def _add_custom_instance(instance_name):
             if instance_name in unknown_instance_ids:
                 instances.add(Instance(id=instance_name, src='jr://fixture/{}'.format(instance_name)))
                 unknown_instance_ids.remove(instance_name)
 
-        if toggles.CUSTOM_CALENDAR_FIXTURE.enabled(self.app.domain):
+        if toggles.CUSTOM_CALENDAR_FIXTURE.enabled(domain):
             _add_custom_instance('enikshay:calendar')
 
-        if toggles.MOBILE_UCR.enabled(self.app.domain):
+        if toggles.MOBILE_UCR.enabled(domain):
             _add_custom_instance('commcare:reports')
 
         instance_name = 'locations'
         if instance_name in unknown_instance_ids:
-            if toggles.FLAT_LOCATION_FIXTURE.enabled(self.app.domain):
+            if toggles.FLAT_LOCATION_FIXTURE.enabled(domain):
                 instances.add(Instance(id=instance_name, src='jr://fixture/{}'.format(instance_name)))
             else:
                 instances.add(Instance(id=instance_name, src='jr://fixture/commtrack:{}'.format(instance_name)))

--- a/corehq/apps/app_manager/suite_xml/sections/remote_requests.py
+++ b/corehq/apps/app_manager/suite_xml/sections/remote_requests.py
@@ -50,16 +50,16 @@ class RemoteRequestFactory(object):
 
     def _build_remote_request_post(self):
         return RemoteRequestPost(
-                url=absolute_reverse('claim_case', args=[self.domain]),
-                relevant=self.module.search_config.relevant,
-                data=[
-                    QueryData(
-                        key='case_id',
-                        ref=QuerySessionXPath('case_id').instance(),
-                        # e.g. instance('querysession')/session/data/case_id
-                    ),
-                ]
-            )
+            url=absolute_reverse('claim_case', args=[self.domain]),
+            relevant=self.module.search_config.relevant,
+            data=[
+                QueryData(
+                    key='case_id',
+                    ref=QuerySessionXPath('case_id').instance(),
+                    # e.g. instance('querysession')/session/data/case_id
+                ),
+            ]
+        )
 
     def _build_command(self):
         return Command(

--- a/corehq/apps/app_manager/tests/test_suite_remote_request.py
+++ b/corehq/apps/app_manager/tests/test_suite_remote_request.py
@@ -31,7 +31,11 @@ class RemoteRequestSuiteTest(SimpleTestCase, TestXmlMixin, SuiteMixin):
             default_properties=[
                 DefaultCaseSearchProperty(
                     property='name',
-                    defaultValue="instance('casedb')/case[@case_id='instance('commcaresession')/session/data/case_id']/some_property"),
+                    defaultValue=(
+                        "instance('casedb')/case"
+                        "[@case_id='instance('commcaresession')/session/data/case_id']"
+                        "/some_property")
+                ),
                 DefaultCaseSearchProperty(
                     property='name',
                     defaultValue="instance('locations')/locations/location[@id=123]/@type",

